### ISSUE TITLE
UIImageView not updating in row after delete

### ISSUE THREAD

**Issue by Moto110** [jdawes/the-beer-recorder-ios#1]
I noticed when a row is deleted the image from the deleted row remains on the screen.  When the row below the deleted row moves up the "deleted" image remains on that row with the title from the item that was below it until the view is refreshed. Any idea what is causing this?  I have tried numerous options to reload the screen after the delete, but since I am still learning I have not had any success.

**Comment by jeffmcnd** [comment]
Hey! I could possibly take a look into this for you. I made this app for a friend and he doesn't even use it so I mostly forgot about it. I'll try to take a look and see if I can explain what's going on.

**Comment by Moto110** [comment]
Thanks!  I have tried everything I can think of to fix it (Basically what Google told me to try).  This code is great for tinkering and learning with the use of beer and beer terms makes it easier for me to follow the code.  What can I say I LOVE beer.

**Comment by jeffmcnd** [comment]
So there a few things that need to be done when updating the table, I've updated the code to show the comments about what's going on but I'll explain here too. The list of things that need to be done is as follows:
1. Delete the saved image from the file system because it is no longer needed.
2. Remove the beer information from the database and update the array that array that holds the beers for the table.
3. Remove the image from the array that holds the images for each row. (This should probably be updated so that you have one array of custom objects that hold the info for the beer and an image or something like that)
4. Reload the table to update all the rows.

Turns out that I had skipped step #3 so every time you delete a row, the images array never got updated so the difference between the beer array and the image array gets worse and worse. Hopefully my recent commit fixes everything. Thanks for filing the issue!
